packages/scheduler/src/forks/Scheduler.js

```diff
@@ -85,34 +85,12 @@ var isHostCallbackScheduled = false;
 var isHostTimeoutScheduled = false;
 
 // Capture local references to native APIs, in case a polyfill overrides them.
-const setTimeout = window.setTimeout;
-const clearTimeout = window.clearTimeout;
-const setImmediate = window.setImmediate; // IE and Node.js + jsdom
-
-if (typeof console !== 'undefined') {
-  // TODO: Scheduler no longer requires these methods to be polyfilled. But
-  // maybe we want to continue warning if they don't exist, to preserve the
-  // option to rely on it in the future?
-  const requestAnimationFrame = window.requestAnimationFrame;
-  const cancelAnimationFrame = window.cancelAnimationFrame;
-
-  if (typeof requestAnimationFrame !== 'function') {
-    // Using console['error'] to evade Babel and ESLint
-    console['error'](
-      "This browser doesn't support requestAnimationFrame. " +
-        'Make sure that you load a ' +
-        'polyfill in older browsers. https://reactjs.org/link/react-polyfills',
-    );
-  }
-  if (typeof cancelAnimationFrame !== 'function') {
-    // Using console['error'] to evade Babel and ESLint
-    console['error'](
-      "This browser doesn't support cancelAnimationFrame. " +
-        'Make sure that you load a ' +
-        'polyfill in older browsers. https://reactjs.org/link/react-polyfills',
-    );
-  }
-}
+const setTimeout =
+  typeof window !== 'undefined' ? window.setTimeout : setTimeout;
```

Comment:
Do we really need to do this?

Reply:
No, we just need to save them early in case they get overriden. So

```js
const localSetTimeout = setTimeout
```

and then use that throughout the file.

Reply:
I think it was only to grab a local reference in case it later gets polyfilled.

So we can do this instead:

```js
const localSetTimeout = setTimeout
const localSetTimeout = clearTimeout
```

No `window` object

Reply:
Should add a test for that

Reply:
Well, only for those that exist everywhere. E.g. we have to do `typeof` check to avoid blowing up on missing `setImmediate`.

Reply:
There was also this issue: https://github.com/facebook/react/pull/13088. Since currently `scheduler` is not needed for SSR, some people load it in `setTimeout`-less environments.

So if you remove `window` here, you might have to write:

```js
let localSetTimeout;
if (typeof setTimeout === 'function') {
  localSetTimeout = setTimeout;
}
```

Otherwise it's a breaking change.

Reply:
Ok, fixed to not use window, but check typeof, and added tests.